### PR TITLE
[2656] - Refactor Client Account Fragment and AccountFragment from XML to Compose

### LINKED ISSUE #2612: refactor: Migrate AccountsFragment from XML to compose

### PR DESCRIPTION
**Pull request by akashmeruva9**
Fixes #2612 

<table style="border: 1px solid black;">
            <tr>
                <td  style="border: 1px solid black ;">
                    
                </td>
                <td  style="border: 1px solid black ;">
                    
                </td>
                <td  style="border: 1px solid black ;">
                    
                </td>
            </tr>
</table>

<table style="border: 1px solid black;">
            <tr>
                <td  style="border: 1px solid black ;">
                    
                </td>
                <td  style="border: 1px solid black ;">
                    
                </td>
                <td  style="border: 1px solid black ;">
                    
                </td>
            </tr>
</table>

<table style="border: 1px solid black;">
            <tr>
                <td  style="border: 1px solid black ;">
                    
                </td>
            </tr>
</table>

https://github.com/openMF/mifos-mobile/assets/106223361/afd1509f-da4a-4790-b1d0-2a0265e8da80

- [x ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsViewModel.kt [hunk 1]
```diff
@@ -30,6 +32,104 @@ class AccountsViewModel @Inject constructor(
...
+//    fun filterAccounts( checkBoxList: List<CheckboxStatus>, accountType: String) {
+//
+//        if(accountType == Constants.SAVINGS_ACCOUNTS){
+//        }else if(accountType == Constants.LOAN_ACCOUNTS) {
+//
+//        }else if(accountType == Constants.SHARE_ACCOUNTS) {
+//
+//        }
+//    }
+
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsViewModel.kt]: remove this commented code.

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsViewModel.kt [hunk 2]
```diff
@@ -30,6 +32,104 @@ class AccountsViewModel @Inject constructor(
...
+        }else if( accountType == Constants.SHARE_ACCOUNTS)
+        {
+            _isRefreshing.value = true
+            loadAccounts(Constants.SHARE_ACCOUNTS)
+        }
+    }
+
+    fun updateSearchQuery(query: String) {
+        _isSearching.update { true }
+        _searchQuery.update { query }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsViewModel.kt]: We use .update when we have to update the data classes of the flows.

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 3]
```diff
@@ -0,0 +1,933 @@
...
+
+                is AccountsUiState.Loading -> {
+                    MifosProgressIndicatorOverlay()
+                }
+
+                is AccountsUiState.ShowLoanAccounts -> {}
+
+                is AccountsUiState.ShowSavingsAccounts -> {
+
+                    if( ( uiState.savingAccounts!!.isEmpty()) )
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: remove extra space after ( and before ) , also remove line

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 4]
```diff
@@ -0,0 +1,933 @@
...
+                        isRetryEnabled = true,
+                        onRetry = onRetry
+                    )
+                }
+
+                is AccountsUiState.Loading -> {
+                    MifosProgressIndicatorOverlay()
+                }
+
+                is AccountsUiState.ShowLoanAccounts -> {}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: is AccountsUiState.ShowLoanAccounts -> Unit

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 5]
```diff
@@ -0,0 +1,933 @@
...
+                            accountsList = uiState.savingAccounts,
+                            isSearching = isSearching,
+                            isFiltered= isFiltered,
+                            getUpdatedSearchList = { accountsList -> getUpdatedSearchList(accountsList) },
+                            getUpdatedFilterList = { accountsList -> getUpdatedFilterList(accountsList) },
+                            onItemClick = onItemClick
+                        )
+                    }
+                }
+                is AccountsUiState.ShowShareAccounts -> {}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: AccountsUiState.ShowShareAccounts -> Unit

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 6]
```diff
@@ -0,0 +1,933 @@
...
+                            isFiltered = isFiltered,
+                            getUpdatedSearchList = { accountsList -> getUpdatedSearchList(accountsList) },
+                            getUpdatedFilterList = { accountsList -> getUpdatedFilterList(accountsList) },
+                            onItemClick = onItemClick
+                        )
+                    }
+                }
+
+                is AccountsUiState.ShowSavingsAccounts -> {}
+                is AccountsUiState.ShowShareAccounts -> {}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: Unit

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 7]
```diff
@@ -0,0 +1,933 @@
...
+                    )
+                }
+
+                is AccountsUiState.Loading -> {
+                    MifosProgressIndicatorOverlay()
+                }
+
+                is AccountsUiState.ShowLoanAccounts -> {}
+
+                is AccountsUiState.ShowSavingsAccounts -> {}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: Unit

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 8]
```diff
@@ -0,0 +1,933 @@
...
+    getUpdatedFilterList: (accountsList: List<SavingAccount?>) -> List<SavingAccount?>,
+    onItemClick: (accountType: String, accountId: Long) -> Unit,
+) {
+
+    var accounts by rememberSaveable {
+        mutableStateOf(accountsList)
+    }
+
+    if( isFiltered && isSearching)
+    {
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: if() {

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 9]
```diff
@@ -0,0 +1,933 @@
...
+    isFiltered: Boolean,
+    getUpdatedFilterList: (accountsList: List<SavingAccount?>) -> List<SavingAccount?>,
+    onItemClick: (accountType: String, accountId: Long) -> Unit,
+) {
+
+    var accounts by rememberSaveable {
+        mutableStateOf(accountsList)
+    }
+
+    if( isFiltered && isSearching)
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: remove space after (

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 10]
```diff
@@ -0,0 +1,933 @@
...
+                }
+
+                is AccountsUiState.ShowLoanAccounts -> {}
+
+                is AccountsUiState.ShowSavingsAccounts -> {}
+
+                is AccountsUiState.ShowShareAccounts -> {
+
+                    if( uiState.shareAccounts!!.isEmpty() )
+                    {
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: { after if()

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 11]
```diff
@@ -0,0 +1,933 @@
...
+        this.approvedString = context?.getString(R.string.approved)
+        this.approvalPendingString = context?.getString(R.string.approval_pending)
+        this.maturedString = context?.getString(R.string.matured)
+        this.waitingForDisburseString = context?.getString(R.string.waiting_for_disburse)
+        this.overpaidString = context?.getString(R.string.overpaid)
+        this.closedString = context?.getString(R.string.closed)
+        this.withdrawnString = context?.getString(R.string.withdrawn)
+        this.inArrearsString = context?.getString(R.string.in_arrears)
+    }
+}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: see how I have done in SavingsAccountTransactionScreen
SavingsTransactionCheckBoxFilter

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 12]
```diff
@@ -0,0 +1,933 @@
...
+    context: Context
+): List<ShareAccount?> {
+    val newList : MutableList<ShareAccount?> = mutableListOf()
+    for( filter in filterList)
+    {
+        if(filter.isChecked)
+            newList.addAll( viewModel.getFilteredShareAccount(accountsList,filter, getFilterStrings(context))!! )
+    }
+    return newList
+}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: this should be in viewmodel

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 13]
```diff
@@ -0,0 +1,933 @@
...
+        AccountsSavingsScreen(
+            uiState = uiState,
+            isSearching = isSearching,
+            isFiltered = isFiltered,
+            navigateBack = { navigateBack },
+            onRetry = { viewModel.loadAccounts(accountType) },
+            isRefreshing = isRefreshing,
+            onRefresh = { viewModel.refresh(accountType) },
+            getUpdatedSearchList = { accountsList -> viewModel.searchInSavingsList(accountsList, searchQuery) },
+            getUpdatedFilterList = { accountsList -> getFilterSavingsAccountList(accountsList = accountsList, filterList, viewModel, context) },
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: are we able to filter the list with this? we are nowhere updating the list

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 14]
```diff
@@ -0,0 +1,933 @@
...
+    getUpdatedFilterList: (accountsList: List<LoanAccount?>) -> List<LoanAccount?>,
+    onItemClick: (accountType: String, accountId: Long) -> Unit,
+) {
+
+    var accounts by rememberSaveable {
+        mutableStateOf(accountsList)
+    }
+
+    if( isFiltered && isSearching)
+    {
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: proper indentation

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 15]
```diff
@@ -0,0 +1,933 @@
...
+                    )
+                } else {
+                    AccountScreenLoanListItem(
+                        loanAccount = loanAccount,
+                        color = colorResource(R.color.gray_dark),
+                        stringResource = "${stringResource(id = R.string.withdrawn)} ${getDateAsString( loanAccount.timeline?.withdrawnOnDate)}",
+                        numcolor = null,
+                        onItemClick = onItemClick
+                    )
+                }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: use when statement

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 16]
```diff
@@ -0,0 +1,933 @@
...
+    if( isFiltered && isSearching)
+    {
+        accounts = getUpdatedSearchList(getUpdatedFilterList( accountsList))
+    }else if( isSearching ){
+        accounts = getUpdatedSearchList(accountsList)
+    }else if( isFiltered ){
+        accounts = getUpdatedFilterList(accountsList)
+    }else {
+        accounts = accountsList
+    }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: use when

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 17]
```diff
@@ -0,0 +1,933 @@
...
+                        )
+                    }
+
+                    shareAccount?.status?.submittedAndPendingApproval == true -> {
+                        AccountScreenShareListItem(
+                            shareAccount = shareAccount,
+                            color = colorResource(R.color.light_yellow),
+                            setSharingAccountDetail = false
+                        )
+
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: remove this unwanted line

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 18]
```diff
@@ -0,0 +1,933 @@
...
+            }
+
+            Text(
+                text = stringResource,
+                style = MaterialTheme.typography.labelLarge,
+                color = colorResource(id = R.color.gray_dark),
+            )
+        }
+
+        Spacer( Modifier.weight(1f))
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: intendation

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 19]
```diff
@@ -0,0 +1,933 @@
...
+                    }
+                }
+            }
+
+
+        }
+
+        Spacer( Modifier.weight(1f))
+
+
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: remove

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt [hunk 20]
```diff
@@ -0,0 +1,933 @@
...
+                lineTo(0f, size.height)
+                close()
+            }
+            drawPath(
+                path = path,
+                color = color
+            )
+        }
+    }
+}
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: move to utils, and use proper naming, like not CustomShapeBox, something more specific

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 21]
```diff
@@ -0,0 +1,421 @@
...
+    val filterList by viewModel.filterList.collectAsStateWithLifecycle()
+    var list = listOf<String>( Constants.SAVINGS_ACCOUNTS, Constants.LOAN_ACCOUNTS, Constants.SHARE_ACCOUNTS )
+
+    LaunchedEffect(key1 = currentPage) {
+        viewModel.setFilterList( emptyList() )
+        getCheckBoxList( currentPage, viewModel, context)
+    }
+
+    if(isDialogActive)
+    {
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: { to prev line

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 22]
```diff
@@ -0,0 +1,421 @@
...
+            currentPage = currentPage )
+    }
+
+    MFScaffold(
+        topBar = { ClientAccountsScreenTopBar(
+            navigateBack = navigateBack,
+            onChange = { viewModel.updateSearchQuery(it) },
+            clickDialog = { isDialogActive = true },
+            closeSearch = { viewModel.stoppedSearching() }
+            )
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: indent

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 23]
```diff
@@ -0,0 +1,421 @@
...
+
+        floatingActionButtonContent = FloatingActionButtonContent(
+            onClick = {
+                when (currentPage) {
+                    0 -> openNextActivity(currentPage)
+                    1 -> openNextActivity(currentPage)
+                }
+            },
+            contentColor = if (isSystemInDarkTheme()) MaterialTheme.colorScheme.secondary else MaterialTheme.colorScheme.primary,
+            content = { Icon(Icons.Filled.Add, contentDescription = "Create Account", tint = if (isSystemInDarkTheme()) Color.White else Color.Black) }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: use MifosIcons

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 24]
```diff
@@ -0,0 +1,421 @@
...
+@Composable
+fun ClientAccountsScreen(
+    navigateBack: () -> Unit?,
+    openNextActivity: (currentPage: Int) -> Unit,
+    onItemClick: (accountType: String, accountId : Long) -> Unit
+)  {
+
+
+    val context = LocalContext.current
+    val viewModel : AccountsViewModel = hiltViewModel()
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: viewModel should be in params, and make another composable for below code

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 25]
```diff
@@ -0,0 +1,421 @@
...
+    var isDialogActive by rememberSaveable { mutableStateOf(false) }
+    var currentPage by remember { mutableIntStateOf(0) }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: rememberSaveable

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 26]
```diff
@@ -0,0 +1,421 @@
...
+
+    if(isDialogActive)
+    {
+        ShowFilterDialog(
+            filterList = filterList,
+            cancelDialog = { isDialogActive = false  },
+            clearFilter = {  getCheckBoxList(currentPage, viewModel, context)
+                            isDialogActive = false  },
+            filter = { filterAccountsIfChanged( checkBoxList =  it, currentPage, viewModel, context )
+                        isDialogActive = false  },
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: indent

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 27]
```diff
@@ -0,0 +1,421 @@
...
+}
+
+fun getCheckBoxList(currentPage: Int, viewModel: AccountsViewModel, context: Context)
+{
+    if(currentPage == 0 )
+         viewModel.setFilterList(StatusUtils.getSavingsAccountStatusList(context))
+    else if( currentPage == 1)
+         viewModel.setFilterList(StatusUtils.getLoanAccountStatusList(context))
+    else if( currentPage == 2 )
+         viewModel.setFilterList(StatusUtils.getShareAccountStatusList(context))
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: when

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 28]
```diff
@@ -0,0 +1,421 @@
...
+
+}
+
+@Composable
+fun ShowFilterDialog(
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: make another file for this

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 29]
```diff
@@ -0,0 +1,421 @@
...
+        }
+
+        HorizontalPager(
+            state = pagerState, modifier = Modifier.fillMaxWidth(), pageContent =  { page ->
+                if( currentPage == 0 )
+                    AccountsScreen(navigateBack =  navigateBack , Constants.SAVINGS_ACCOUNTS, onItemClick)
+                else if( currentPage == 1)
+                    AccountsScreen(navigateBack = navigateBack, Constants.LOAN_ACCOUNTS, onItemClick )
+                else if( currentPage == 2 )
+                    AccountsScreen(navigateBack = navigateBack, Constants.SHARE_ACCOUNTS, onItemClick )
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: when

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt [hunk 30]
```diff
@@ -0,0 +1,421 @@
...
+        pageChanged(currentPage)
+        pagerState.animateScrollToPage(currentPage)
+    }
+
+    LaunchedEffect(key1 = pagerState.currentPage, pagerState.isScrollInProgress) {
+        if(!pagerState.isScrollInProgress)
+        currentPage = pagerState.currentPage
+        else {
+            currentPage = pagerState.targetPage
+        }
```
Comment (AvneetSingh2001) [inline app/src/main/java/org/mifos/mobile/ui/client_accounts/ClientAccountsScreen.kt]: indent

**Review by AvneetSingh2001 — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt (hunk 13 above)
Comment (akashmeruva9) [reply app/src/main/java/org/mifos/mobile/ui/client_accounts/AccountsScreen.kt]: these functions filter the list and return a new list into the parameter

**Pull request merged**